src/openrct2/peep/Peep.cpp

```diff
@@ -3068,22 +3068,55 @@ void Peep::PerformNextAction(uint8_t& pathing_result, TileElement*& tile_result)
 
     auto newLoc = *loc;
     CoordsXY truncatedNewLoc = newLoc.ToTileStart();
+    int16_t height = GetZOnSlope(newLoc.x, newLoc.y);
     if (truncatedNewLoc == CoordsXY{ NextLoc })
     {
-        int16_t height = GetZOnSlope(newLoc.x, newLoc.y);
         MoveTo(newLoc.x, newLoc.y, height);
         return;
     }
 
-    if (newLoc.x < 32 || newLoc.y < 32 || newLoc.x >= gMapSizeUnits || newLoc.y >= gMapSizeUnits)
+
+    bool outside = false;
+    if (newLoc.x <= 32 || newLoc.y <= 32 || newLoc.x >= gMapSizeUnits || newLoc.y >= gMapSizeUnits)
+        // apply same old behaviour
+        outside = true;
```

Comment:
Please don't use if's like this. Its a classic way of falling into the goto fail trap. https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/